src/output-json-anomaly.c

```diff
@@ -351,7 +366,15 @@ static void JsonAnomalyLogConf(AnomalyJsonOutputCtx *json_output_ctx,
 {
     uint16_t flags = 0;
     if (conf != NULL) {
-        SetFlag(conf, "packethdr", LOG_JSON_PACKETHDR, &flags);
+        SetFlag(conf, "protoparser", LOG_JSON_PROTOPARSER, &flags);
+        SetFlag(conf, "parser",      LOG_JSON_PARSER,      &flags);
+        SetFlag(conf, "protodetect", LOG_JSON_PROTODETECT, &flags);
+        SetFlag(conf, "protodecode", LOG_JSON_PROTODECODE, &flags);
+        SetFlag(conf, "packethdr",   LOG_JSON_PACKETHDR,   &flags);
+    }
+    if (flags == 0) {
+        SCLogNotice("Anomaly logging has been configured; however, no logging choices "
```

Comment:
when I enable the logger in the yaml, but don't enable any of the types, the notice message appears twice

Reply:
This can be confusing ... the anomaly logger registers 2 loggers - `OutputRegisterPacketSubModule` and `OutputRegisterTxSubModule`

Reply:
I'll ensure that the message is only displayed once, even if there are multiple logging instances.